### PR TITLE
[rework] Inventory: Make Item class NOT a child of interactable class 

### LINKED ISSUE #342: Inventory: Make Item class NOT a child of interactable class

### PR DESCRIPTION
**Pull request by VadimBhirawa**
### Main changes
Interactable.cs:
- AudioPlayerObj dijadiin private
- PlayAudio(string name) dijadiin private

Item.cs:
- Tambahin AudioPlayerObj dan PlayAudio (sama kyk punya Interactable.cs), dibikin private juga
- Rename OnInteraction() jadi Pick() (pake ctrl+r+r, biarin aja error)
- Apus fungsi OnInteraction() yg barusan direname jadi Pick(), Pick() yg lama biarin aja
- Set Pick yg lama jadi public
- Apus interface IItem dan bikin interface baru (autogenerate aja)

Detectors:
- Pindahin semua fungsi dan variable di ObjectDetector.cs ke InteractableDetector.cs dan ItemDetector.cs terus sesuain tipenya (InteractableDetector buat Interactable aja, ItemDetector buat Item aja)
- Delete class ObjectDetector.cs
- Ilangin inheritance InteractableDetector.cs dan ItemDetector.cs ke ObjectDetector.cs
- GhostSimulationInteractableDetector.cs inherits ItemDetector.cs

How to check:
- Check the codes
- Run tests

### Tasks List
- [close #342 ]

### CODE REVIEW

**Review by InShineStudios — CHANGES_REQUESTED** [review]
Since Item is no longer a subclass of Interactable, the foldering should also change. Move the "Items" folder so that it is on the same level as the "Interactables" folder.

Adjust foldering as well for the Interactables and Items scripts:
Make folders by object name. For example folder named "Door" and inside are "DoorAnimator" and "DoorController" scripts. Similarly, do the same for all interactables (light switch, victim, closets). Also: rename "ClosetsController" to "ClosetController"

For items:
Make folder "ExorcismItem" and move the script to that folder. The other item scripts should be fine.

File: Assets/Astralization/Scripts/Player/Detectors/GhostSimulationInteractableDetector.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@
  * Will call the onGhostInteraction function on the Interactable object.
  */
 [RequireComponent(typeof(CapsuleCollider))]
-public class GhostSimulationInteractableDetector : ObjectDetector
+public class GhostSimulationInteractableDetector : ItemDetector
```
[the remark this answers is not in the document]
Comment (rafiftaris) [reply Assets/Astralization/Scripts/Player/Detectors/GhostSimulationInteractableDetector.cs:10]: kalo sekarang fungsionalitasnya masih kasar (cuma dipake buat ngetes evidence item), kalo udah fix cara ghost interact dengan environment baru class ini bisa diexpand

File: Assets/Astralization/Scripts/Player/Detectors/InteractableDetector.cs [hunk 2]
```diff
@@ -7,22 +8,114 @@
  * Will call the OnInteraction function on the Interactable object.
  */
 [RequireComponent(typeof(CapsuleCollider))]
-public class InteractableDetector : ObjectDetector
+public class InteractableDetector : MonoBehaviour
```
[the remark this answers is not in the document]
Comment (rafiftaris) [reply Assets/Astralization/Scripts/Player/Detectors/InteractableDetector.cs:11]: betull

File: Assets/Astralization/Scripts/Player/Detectors/InteractableDetector.cs [hunk 3]
```diff
@@ -7,22 +8,114 @@
...
+    #region SetGet
+    public Interactable GetClosest()
+    {
+        return closestInteractable;
+    }
+    #endregion
+
     #region MonoBehaviour
-    protected override void Awake()
+    protected virtual void Awake()
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Player/Detectors/InteractableDetector.cs:33]: not refered by any other class, set to private and remove virtual

File: Assets/Astralization/Scripts/Player/Detectors/InteractableDetector.cs [hunk 4]
```diff
@@ -7,22 +8,114 @@
...
+
+    private void OnDisable()
+    {
+        PlayerMovement.FindClosest -= UpdateClosestInteractable;
     }
     #endregion
 
     #region Handler
-    protected override void InteractClosest(Interactable closest)
+    protected virtual void InteractClosest(Interactable closest)
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Player/Detectors/InteractableDetector.cs:78]: not refered by any other class, set to private and remove virtual

File: Assets/Astralization/Scripts/Player/Detectors/InteractableDetector.cs [hunk 5]
```diff
@@ -7,22 +8,114 @@
...
     {
         //Debug.Log("[PLAYER INTERACTION] Interacted with " + closest.name);
         closest.OnInteraction();
         UpdateClosestInteractable();
     }
+
+    protected void UpdateClosestInteractable()
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Player/Detectors/InteractableDetector.cs:85]: set to private

File: Assets/Astralization/Scripts/Player/Detectors/ItemDetector.cs [hunk 6]
```diff
@@ -7,30 +8,120 @@
...
+    #region SetGet
+    public Item GetClosest()
+    {
+        return closestItem;
+    }
     #endregion
 
     #region MonoBehaviour
-    protected override void Awake()
+    protected virtual void Awake()
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Player/Detectors/ItemDetector.cs:36]: not refered by any other class, set to private and remove virtual

File: Assets/Astralization/Scripts/Player/Detectors/ItemDetector.cs [hunk 7]
```diff
@@ -7,30 +8,120 @@
...
         //Debug.Log("[PLAYER INTERACTION] Picked " + closest.name);
         _inventory.PickItem((Item)closest);
         nearbyInteractables.Remove(closest);
-        closestInteractable = null;
-        UpdateClosestInteractable();
+        closestItem = null;
+        UpdateClosestItem();
+    }
+
+    protected void UpdateClosestItem()
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Player/Detectors/ItemDetector.cs:91]: set to private

**Review by rafiftaris — CHANGES_REQUESTED** [review]
- Delete ObjectDetector.cs script

File: Assets/Astralization/Scripts/Player/Detectors/ItemDetector.cs (hunk 6 above)
Comment (VadimBhirawa) [reply Assets/Astralization/Scripts/Player/Detectors/ItemDetector.cs:36]: this one is refered by GhostSimulationInteractableDetector so I don't think it should be changed to private(?) (Error showed up after this was changed to private)

Comment (rafiftaris) [reply Assets/Astralization/Scripts/Player/Detectors/ItemDetector.cs:36]: oh okay lol, keep it that way then

**Review by InShineStudios — APPROVED** [review]
Nice refactoring

**Review by rafiftaris — APPROVED** [review]
mantapp bro!

**Pull request merged**